### PR TITLE
feat: Adding Apache Kafka Kubernetes Add-on

### LINKED ISSUE #1089: Add Apache Kafka Kubernetes Add-on
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

#### What is the outcome that you are trying to reach?

Adding Kafka Operator available to use as Kafka add-on

#### Describe the solution you would like

Use Strimzi Operator as a Kubernetes add-on to be used to install Kafka Clusters

#### Describe alternatives you have considered

The first Alternative is to install Confluent for Kubernetes, but favored the opensource solution for the first step and we can add confluent later. Also confluent comes with more components that may not be required.

We discussed adding the add-on in the DoEKS repo, but preferred to keep the add-on (operator) here and the complete solution to be implemented in DoEKS repo.

#### Additional context

This is a per-requisite step to this issue : https://github.com/awslabs/data-on-eks/issues/22

### PR DESCRIPTION
**Pull request by AlyIbrahim**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Add Strmizi Operator as a Kubernetes Add-on to be used by DoEKS for Apache Kafka on EKS

### Motivation

- Resolves #1089 
- Pr-requisite for https://github.com/awslabs/data-on-eks/issues/22 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Comment by AlyIbrahim** [comment]
@vara-bonthu

**Review by vara-bonthu — APPROVED** [review]
LGTM! I have fixed few formatting issues and added doc

Let's wait for one more approval @bryantbiggs , @askulkarni2

File: docs/add-ons/kafka.md
Comment (askulkarni2) [inline docs/add-ons/kafka.md:1]: Strimzi Kafka Operator?

File: modules/kubernetes-addons/kafka/values.yaml
Comment (bryantbiggs) [inline modules/kubernetes-addons/kafka/values.yaml:13]: We should remove this since they are the chart defaults. It makes version changes easier for users 

https://github.com/strimzi/strimzi-kafka-operator/blob/main/helm-charts/helm3/strimzi-kafka-operator/values.yaml

**Review by bryantbiggs — COMMENTED** [review]
Looks good - let's only add values that are non-default to make version changes easier for users 

We should also get this added to the GitOps addon repo

File: modules/kubernetes-addons/kafka/values.yaml
Comment (vara-bonthu) [reply modules/kubernetes-addons/kafka/values.yaml]: I removed the defaults except the resources section which are modified

**Comment by vara-bonthu** [comment]
@AlyIbrahim  Would you be able to raise another PR for ArgoCD deployment? Here is the [repo](https://github.com/aws-samples/eks-blueprints-add-ons) location. Please reach out if you need help.

**Review by askulkarni2 — APPROVED** [review]
LGTM!

**Comment by AlyIbrahim** [comment]
@vara-bonthu Yep, will check that

**Review by AlyIbrahim — COMMENTED** [review]
LGTM

**Pull request merged**
